url888b6fd9b4_jmacglashan_commandsToTasks_tgz-pJ8-commands_tests_Model3TestsJ8/src/domain/singleagent/sokoban/SokobanDomain.java

```diff
@@ -0,0 +1,1345 @@
+		for(int i = 0; i < n; i++){
+			ObjectInstance block = new ObjectInstance(SOKOBANDOMAIN.getObjectClass(BLOCKCLASS), BLOCKCLASS + i);
+			blocks.add(block);
+		}
+
+		State st = new State();
+		st.addObject(room);
+		st.addObject(room2);
+		st.addObject(room3);
+		
+		for(ObjectInstance block : blocks){
+			st.addObject(block);
+		}
+		
+		st.addObject(agent);
+		st.addObject(door);
+		st.addObject(door2);
+
+		return st;
+		
+	}
+	
+	public State getCleanStateNBlocksNDoors(int nb, int nd){
+		
+		this.generateDomain(); //make sure the domain is created first
+
+		ObjectInstance agent = new ObjectInstance(SOKOBANDOMAIN.getObjectClass(AGENTCLASS), AGENTCLASS + 0);
+		ObjectInstance room = new ObjectInstance(SOKOBANDOMAIN.getObjectClass(ROOMCLASS), ROOMCLASS + 0);
+		ObjectInstance room2 = new ObjectInstance(SOKOBANDOMAIN.getObjectClass(ROOMCLASS), ROOMCLASS + 1);
+		ObjectInstance room3 = new ObjectInstance(SOKOBANDOMAIN.getObjectClass(ROOMCLASS), ROOMCLASS + 2);
+		
+		
+		
+		List <ObjectInstance> doors = new ArrayList<ObjectInstance>();
+		for(int i = 0; i < nd; i++){
+			ObjectInstance door = new ObjectInstance(SOKOBANDOMAIN.getObjectClass(DOORCLASS), DOORCLASS + i);
+			doors.add(door);
+		}
+		
+		List <ObjectInstance> blocks = new ArrayList<ObjectInstance>();
+		for(int i = 0; i < nb; i++){
+			ObjectInstance block = new ObjectInstance(SOKOBANDOMAIN.getObjectClass(BLOCKCLASS), BLOCKCLASS + i);
+			blocks.add(block);
+		}
+
+		State st = new State();
+		st.addObject(room);
+		st.addObject(room2);
+		st.addObject(room3);
+		
+		for(ObjectInstance block : blocks){
+			st.addObject(block);
+		}
+		
+		st.addObject(agent);
+		
+		for(ObjectInstance door : doors){
+			st.addObject(door);
+		}
+
+		return st;
+		
+		
+		
+	}
+
+	//creates the walls using the coordinates of rooms
+	public static void createMap(State st){
+
+		MAXX = getMaxX(st);
+		MAXY = getMaxY(st);
+		
+		MAP = new int[MAXX+1][MAXY+1];
+		
+		//start by zeroing it out
+		for(int i = 0; i < MAXX; i++){
+			for(int j = 0; j < MAXY; j++){
+				MAP[i][j] = 0;
+			}
+		}
+
+		List<ObjectInstance> rooms = st.getObjectsOfTrueClass(ROOMCLASS);
+		List<ObjectInstance> doors = st.getObjectsOfTrueClass(DOORCLASS);
+
+		for (int i = 0; i < rooms.size(); i++){
+			ObjectInstance room = rooms.get(i);
+
+			int topX = room.getDiscValForAttribute(TOPXATTNAME);
+			int topY = room.getDiscValForAttribute(TOPYATTNAME);
+			int botX = room.getDiscValForAttribute(BOTTOMXATTNAME);
+			int botY = room.getDiscValForAttribute(BOTTOMYATTNAME);
+
+			//put walls at the edges of rooms
+			for(int j = 0; j < MAXX; j++){
+				for(int k = 0; k < MAXY; k++){
+					if (j >= topX && j <= botX && k == topY){
+						MAP[j][k] = 1;
+					}
+					else if (j >= topX && j <= botX && k == botY){
+						MAP[j][k] = 1;
+					}
+					else if (j == topX && k >= botY && k <= topY){
+						MAP[j][k] = 1;
+					}
+					else if (j == botX && k >= botY && k <= topY){
+						MAP[j][k] = 1;
+					}
+				}
+			}
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 601:626, 629:653.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.